### PR TITLE
change interface to prune, and added more tests

### PR DESCRIPTION
**Pull request by lducas**
Sorry for to API def war... But I think this makes more sense.

### CODE REVIEW

**Comment by lducas** [comment]
Note 1 : in sync with fplll 's branch pruner. 
Note 2: the greedy method also returns an updated radius (lesser or equal to the original).

File: src/fpylll/fplll/pruner.pyx [hunk 1]
```diff
@@ -106,32 +105,44 @@ def prune(pruning, double enumeration_radius, double preproc_cost, double target
...
         prune_c[FP_NR[double]]((<Pruning>pruning)._core, enumeration_radius, preproc_cost, target, vec, descent_method, metric, reset)
         sig_off()
+        if descent_method==PRUNER_METHOD_GREEDY:
+            return (enumeration_radius, pruning)
         return pruning
     if ft == FT_LONG_DOUBLE:
         sig_on()
         prune_c[FP_NR[longdouble]]((<Pruning>pruning)._core, enumeration_radius, preproc_cost, target, vec, descent_method, metric, reset)
         sig_off()
+        if descent_method==PRUNER_METHOD_GREEDY:
```
Comment (malb) [inline src/fpylll/fplll/pruner.pyx]: `==` → ` == ` (spaces) :)

File: src/fpylll/fplll/pruner.pyx [hunk 2]
```diff
@@ -106,32 +105,44 @@ def prune(pruning, double enumeration_radius, double preproc_cost, double target
...
+            return (enumeration_radius, pruning)
```
Comment (malb) [inline src/fpylll/fplll/pruner.pyx]: You don't need those brackets

File: tests/test_pruner.py [hunk 3]
```diff
@@ -1,31 +1,81 @@
...
-            vec.append(tuple(dump_r(m, 0, n)))
+def test_pruner_vec():
+    for (n, overhead) in dim_oh:
 
-    radius = sum([mat.get_r(0, 0) for mat in M])/len(M)
-    pruning = prune(None, radius, 0, 0.9, vec)
-    assert pruning.expectation >= 0.89
-    pruning = prune(pruning, radius, 0, 0.9, vec, reset=False)
-    assert pruning.expectation >= 0.89
+        print " \n ==== Dim \n", n
```
Comment (malb) [inline tests/test_pruner.py]: This isn't Python3 compatible =print foo= → =print(foo)=

**Pull request merged**